src/lib/utils/concepts.h

```diff
@@ -33,14 +33,26 @@ struct is_strong_type<Strong<Ts...>> : std::true_type {};
 template <typename... Ts>
 constexpr bool is_strong_type_v = is_strong_type<std::remove_const_t<Ts>...>::value;
 
-template <typename T0, typename... Ts>
+template <typename T0 = void, typename... Ts>
 struct all_same {
-      static constexpr bool value = (std::is_same_v<T0, Ts> && ...);
+      static constexpr bool value = (std::is_same_v<T0, Ts> && ... && true);
```

Comment:
What's up with this?

Reply:
That's a default value, if the parameter pack (`T0`) is empty.